datajoint/autopopulate.py

```diff
@@ -173,8 +173,7 @@ def populate(
         :param limit: if not None, check at most this many keys
         :param max_calls: if not None, populate at most this many keys
         :param display_progress: if True, report progress_bar
-        :param processes: number of processes to use. When set to a large number, then
-            uses as many as CPU cores
+        :param processes: number of processes to use. Set to None to use all cores
```

Comment:
This is a great idea to improve how this can be specified but doesn't seem to be true based on what is here. 

If you do in fact set `process=None` the following would happen:
- [216]: Should throw an error like: `TypeError: '>' not supported between instances of 'NoneType' and 'int'`

You could probably just replace that particular if-block with something like this:

`min(nkeys, *([processes, mp.cpu_count()] if processes else [mp.cpu_count()]))`

OR

`min(*(_ for _ in (processes, nkeys, mp.cpu_count()) if _))`

These approaches would leverage argument expansion to our benefit to expand only the relevant values into the `min` function.